EIPS/eip-3074.md

```diff
@@ -170,6 +194,16 @@ As with `CALL`, the full gas cost is charged immediately, independently of actua
 
 ## Rationale
 
+### Signature in Memory
+
+The signature format (`yParity`, `r`, and `s`) is fixed, so it might seem curious that `auth` accepts a dynamic memory range. The signature is placed in memory so that `auth` can be upgraded in the future to work with contract accounts (which might use non-ECDSA signatures) and not just EOAs.
+
+### Signing Address `auth` Argument
+
+Including `authority` (the signing address) as an argument to `auth` allows future upgrades to the instruction to work with contract accounts, and not just EOAs.
+
+If `authority` were not included and multiple signature schemes allowed, it would not be possible to compute the authorizing account's address from just the signature alone.
```

Comment:
I'm no expert here, this does make sense, but if there is a signing scheme which would also return 20-byte addresses then it is not known by the address format which signing scheme should be used. I'm not sure if this is a limitation?

Reply:
You're totally right. If you don't know the scheme, you don't know how to derive the address. That's why you have to pass in `authority` into `AUTH`.